tests/Invoke-DbaDiagnosticQuery.Tests.ps1

```diff
@@ -1,16 +1,98 @@
 Write-Host -Object "Running $PSCommandpath" -ForegroundColor Cyan
 . "$PSScriptRoot\constants.ps1"
 
+
 Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
-    Context "Verifying output" {
+    BeforeAll {
+        $script:PesterOutputPath = "TestDrive:$commandName"
+
+
+        $database = "dbatoolsci_frk_$(Get-Random)"
+        $database2 = "dbatoolsci_frk_$(Get-Random)"
+        $server = Connect-DbaInstance -SqlInstance $script:instance2
+        $server.Query("CREATE DATABASE [$database]")
+        $server.Query("CREATE DATABASE [$database2]")
+
+    }
+    AfterAll {
+        $null = Get-DbaDatabase -SqlInstance $server -Databases @($database, $database2) | Remove-DbaDatabase -Confirm:$false
+        Remove-Item $script:PesterOutputPath -recurse -erroraction silentlycontinue
+
+    }
+    AfterEach {
+        Remove-Item $script:PesterOutputPath -Recurse -erroraction silentlycontinue
+    }
+
+
+    Context "verifying output when running queries" {
         It "runs a specific query" {
-            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage' *>&1
-            $results.Name.Count | Should Be 1
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage'
+            @($results).Count | Should -Be 1
         }
         It "works with DatabaseSpecific" {
-            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific *>&1
-            $results.Name.Count -gt 10 | Should Be $true
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific
+            @($results).Count | Should -BeGreaterThan 10
+        }
+        It "works with Exclude Databases provided" {
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific -ExcludeDatabase $database2
+            @($results | Where-Object {$_.DatabaseName -eq $Database1}).Count | Should -BeGreaterThan 1
+            @($results | Where-Object {$_.DatabaseName -eq $Database2}).Count | Should -Be 0
+        }
+    }
+
+    context "verifying output when exporting queries as files instead of running" {
+
+        It "exports queries to sql files without running" {
+            Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -ExportQueries -QueryName 'Memory Clerk Usage' -OutputPath $script:PesterOutputPath
+            @(Get-ChildItem -path $script:PesterOutputPath -filter *.sql).Count | Should -Be 1
+        }
+
+        It "verifies returns object with required properties when running with whatif" {
+            [System.Management.Automation.PSObject[]]$results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -WhatIf
+
+            @($results).count                                                             | Should -BeGreaterThan 10
+            (($results).ComputerName     | Where-Object {$_ -eq $null}).count             | Should -Be 0
```

Comment:
why all those parenthesis ? `$results.ComputerName` isn't the same of `($results).ComputerName`

Reply:
Sorry, I'm unclear. I will address if my next pull request gets this wrong. I adjusted. I also couldn't be sure I was clear on why it impacted result. I got the same results when reviewing

```powershell
($results).Name.GetType() 
$results.Name.GetTYpe() 
($null).Count
$null.Count
```